lib/array_equals.rb

```diff
@@ -1,5 +1,28 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+  
```

Comment:
You should check to see if the arrays are both `nil` first.  You can't for example do `.each` on a `nil.